drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -372,16 +410,91 @@ static int sdw_compute_bus_params(struct sdw_bus *bus)
 	}
 
 	if (i == clk_values) {
-		dev_err(bus->dev, "%s: could not find clock value for bandwidth %d\n",
+multilane:
+		dev_dbg(bus->dev, "%s: could not find clock value for bandwidth %d, checking multi-lane\n",
 			__func__, bus->params.bandwidth);
-		return -EINVAL;
+
+		m_rt = list_last_entry(&bus->m_rt_list, struct sdw_master_runtime, bus_node);
+		/*
+		 * Find available manager lanes that connected to the first Peripheral. No need
+		 * to check all Peripherals available lanes because we can't use multi-lane
+		 * if we can't find any available lane for the first Peripheral.
+		 */
+		s_rt = list_first_entry(&m_rt->slave_rt_list, struct sdw_slave_runtime, m_rt_node);
+		slave_prop = &s_rt->slave->prop;
+
+		for (l = 1; l < SDW_MAX_LANES; l++) {
+			if (!slave_prop->lane_maps[l])
+				continue;
+
+			required_bandwidth = 0;
+			list_for_each_entry(m_p_rt, &m_rt->port_list, port_node) {
+				required_bandwidth += m_rt->stream->params.rate *
+					hweight32(m_p_rt->ch_mask) *
+					m_rt->stream->params.bps;
+			}
+			if (required_bandwidth <= curr_dr_freq - bus->lane_used_bandwidth[l]) {
+				/* Check if m_lane is connected to all Peripherals */
+				if (!check_all_peripherals_connected(m_rt, slave_prop->lane_maps[l])) {
+					dev_dbg(bus->dev,
+						"some Peripherals are not connected to %d\n",
+						slave_prop->lane_maps[l]);
+					continue;
+				}
+				m_lane = slave_prop->lane_maps[l];
+				dev_dbg(&s_rt->slave->dev,
+					"M lane %d P lane %d can be used\n",
+					m_lane, l);
+				bus->lane_used_bandwidth[l] += required_bandwidth;
+				/*
+				 * Use non-zero manager lane, subtract the lane 0
+				 * bandwidth that is already calculated
+				 */
+				bus->params.bandwidth -= required_bandwidth;
+				use_multi_lane = true;
+				break;
+			}
+		}
+
+		if (!use_multi_lane) {
+			dev_err(bus->dev, "%s: could not find an available lane\n", __func__);
```

Comment:
Probably better wording of the error is:
`"%s: multilane is not supported and no clock value found for bandwidth %d\n"` ?

Reply:
> Probably better wording of the error is: `"%s: multilane is not supported and no clock value found for bandwidth %d\n"` ?

We may not find an available lane when multilane is supported. For example, a codec connect lane 0 and lane 1, but the lane bandwidth is currently used by other codecs.

Reply:
Right, but I think the omission of the bandwidth target from the error print is not good, now the error can mean several things and from the log we will have no clue about the reason.

Reply:
I got your point now. Updated to `"%s: multilane is not available and could not find clock value for bandwidth %d\n"`